lib/request_handler/multipart_parser.rb

```diff
@@ -35,8 +35,11 @@ def parse_part(name)
     end
 
     def parse_data(name)
-      JsonApiDocumentParser.new(
-        document:         load_json(name),
+      data = load_json(name)
+      type = multipart_config.dig(name.to_sym, :type)
```

Comment:
is `multipart_config` a hash or some extended Hashie thingy?  if it's the former, this may break on Ruby versions older than 2.3 as `Hash#dig` was introduced in that version